Adding creation_time and last_update_time to experiments

subramaniam02 opened the pull request:
Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

## Related Issues/PRs

<!--
Please reference any related feature requests, issues, or PRs here. For example, `#123`. To automatically close the referenced items when this PR is merged, please use a closing keyword (close, fix, or resolve). For example, `Close #123`. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for more information.
-->
closes #6462 

## What changes are proposed in this pull request?

- creation_time and last_update_time added to experiments
     - changes to File Store
     - changes to Sqlalchemy Store
     - sqlalchmery schema changes


## How is this patch tested?

<!--
If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask. We'd be happy to help.
-->
- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

<!--
Please describe how you confirmed the proposed feature/bug-fix/change works here. For example, if you fixed an MLflow client API, you could attach the code that didn't work prior to the fix but works now, or if you added a new feature on MLflow UI, you could attach a video that demonstrates the feature.
-->

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

added `creation_time` and `last_update_time` for experiments in mlflow python client

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->
<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

jinzhang21 committed:
Changed the name Start Time to Created (#6597)

Signed-off-by: Jin Zhang <jin.zhang@databricks.com>

Signed-off-by: Jin Zhang <jin.zhang@databricks.com>
Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
adding creation_time and last_update_time to experiments

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
adding creation_time and last_update_time to experiments

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
removed tuple

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
removed tuple

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
DB schema

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
DB schema

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
last_update_time addition to experiments file_store

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
last_update_time addition to experiments file_store

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
fixed unit test

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
fixed unit test

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
updating latest schema

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
updating latest schema

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
schema changes in /tests

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
schema changes in /tests

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
examples updated in fluent.py

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
examples updated in fluent.py

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
adding missing DB schema step

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
adding missing DB schema step

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
renaming alembic version file

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
renaming alembic version file

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 commented on tests/db/test_schema.py:
```diff
@@ -118,7 +118,7 @@ def initialize_database():
 
 
-def get_schema_udpate_command(dialect):
+def get_schema_update_command(dialect):
```
Fixed typo !

subramaniam02 commented on tests/db/update_schemas.sh:
```diff
@@ -1,6 +1,8 @@
 #!/bin/bash
 set -ex
 
+python tests/store/dump_schema.py tests/resources/db/latest_schema.sql
```
This seems to be a required step for schema migration of sqlalchemy stores. This exact step is missing in https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#database-schema-changes 
- adding this step to existing DB migration script

harupy replied:
Nice!

subramaniam02 commented on tests/db/compose.yml:
```diff
@@ -47,7 +47,7 @@ services:
     restart: always
     environment:
-      ACCEPT_EULA: Y
+      ACCEPT_EULA: "Y"
```
https://github.com/microsoft/mssql-docker/issues/199

harupy replied:
Why do we need double quotes? `ACCEPT_EULA: Y` is equivalent to `ACCEPT_EULA: "Y"`, isn't it?

subramaniam02 commented:
@harupy Looking forward to your PR review :)

harupy commented on mlflow/store/tracking/dbmodels/models.py:
```diff
@@ -71,6 +71,9 @@ class SqlExperiment(Base):
     Lifecycle Stage of experiment: `String` (limit 32 characters).
                                     Can be either ``active`` (default) or ``deleted``.
     """
+    creation_time = Column(BigInteger(), default=int(time.time()))
+
+    last_update_time = Column(BigInteger(), default=int(time.time()))
```
```suggestion
    creation_time = Column(BigInteger(), default=int(1000 * time.time()))

    last_update_time = Column(BigInteger(), default=int(1000 * time.time()))
```

harupy replied:
Can you add docstring?

harupy commented on mlflow/store/tracking/file_store.py:
```diff
@@ -337,7 +337,14 @@ def _create_experiment_with_id(self, name, experiment_id, artifact_uri, tags):
+            name,
+            artifact_uri,
+            LifecycleStage.ACTIVE,
+            creation_time=int(time.time()),
+            last_update_time=int(time.time()),
```
```suggestion
            creation_time=int(1000* time.time()),
            last_update_time=int(1000 * time.time()),
```

harupy commented on mlflow/store/tracking/file_store.py:
```diff
@@ -450,6 +465,14 @@ def restore_experiment(self, experiment_id):
         mv(experiment_dir, self.root_directory)
+        experiment = self._get_experiment(experiment_id)
+        meta_dir = os.path.join(self.root_directory, experiment_id)
+        experiment._set_last_update_time(int(time.time()))
```
```suggestion
        experiment._set_last_update_time(int(1000 * time.time()))
```

harupy commented on mlflow/store/tracking/file_store.py:
```diff
@@ -463,6 +486,7 @@ def rename_experiment(self, experiment_id, new_name):
             )
         self._validate_experiment_does_not_exist(new_name)
         experiment._set_name(new_name)
+        experiment._set_last_update_time(int(time.time()))
```
```suggestion
        experiment._set_last_update_time(int(time.time() * 1000))
```

harupy commented on mlflow/entities/experiment.py:
```diff
@@ -67,6 +97,10 @@ def to_proto(self):
         experiment.name = self.name
         experiment.artifact_location = self.artifact_location
         experiment.lifecycle_stage = self.lifecycle_stage
+        if self.creation_time:
+            experiment.creation_time = self.creation_time
+        if self.last_update_time:
+            experiment.last_update_time = self.last_update_time
```
Do we need `if`?

```suggestion
        experiment.creation_time = self.creation_time
        experiment.last_update_time = self.last_update_time
```

subramaniam02 replied:
@harupy here is the error without if checks. To be backward compatible, we need these checks

```
Traceback (most recent call last):
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/flask/app.py", line 1822, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/flask/app.py", line 1820, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/flask/app.py", line 1796, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/mlflow/server/handlers.py", line 454, in wrapper
    return func(*args, **kwargs)
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/mlflow/server/handlers.py", line 524, in wrapper
    return func(*args, **kwargs)
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/mlflow/server/handlers.py", line 980, in _list_experiments
    response_message.experiments.extend([e.to_proto() for e in experiment_entities])
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/mlflow/server/handlers.py", line 980, in <listcomp>
    response_message.experiments.extend([e.to_proto() for e in experiment_entities])
  File "/Users/subras/github/mlflow-trial/local/lib/python3.7/site-packages/mlflow/entities/experiment.py", line 101, in to_proto
    experiment.creation_time = self.creation_time
TypeError: 'NoneType' object cannot be interpreted as an integer
```

harupy replied:
Thanks for sharing the traceback. Got it, let's keep the current approach.

subramaniam02 committed:
time.time() * 1000

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
time.time() * 1000

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 commented:
@harupy Not sure if we want to update this line https://github.com/mlflow/mlflow/blob/08205116b0844d43fdeef5062ff1d81304cce17a/mlflow/store/tracking/dbmodels/models.py#L138

harupy commented:
@subramaniam02 I think this line should also be fixed.

subramaniam02 committed:
revert accept_yula

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
revert accept_yula

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
int(time.time() * 1000) for sqlruns

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
int(time.time() * 1000) for sqlruns

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

harupy commented on mlflow/store/tracking/dbmodels/models.py:
```diff
@@ -71,6 +71,14 @@ class SqlExperiment(Base):
     """
+    creation_time = Column(BigInteger(), default=int(time.time() * 1000))
+    """
+    Creation Time of experiment: `BigInteger`.
```
```suggestion
    Creation time of experiment: `BigInteger`.
```

harupy commented on mlflow/store/tracking/dbmodels/models.py:
```diff
@@ -71,6 +71,14 @@ class SqlExperiment(Base):
+    """
+    last_update_time = Column(BigInteger(), default=int(time.time() * 1000))
+    """
+    Last Update Time of experiment: `BigInteger`.
```
```suggestion
    Last Update time of experiment: `BigInteger`.
```

harupy commented on tests/entities/test_experiment.py:
```diff
@@ -1,50 +1,67 @@
         name = "exp_%d_%d" % (random_int(), random_int())
         lifecycle_stage = LifecycleStage.ACTIVE
         location = random_file(".json")
+        creation_time = int(time.time())
+        last_update_time = int(time.time())
```
```suggestion
        creation_time = int(time.time() * 1000)
        last_update_time = int(time.time() * 1000)
```

harupy commented on tests/entities/test_experiment.py:
```diff
@@ -1,50 +1,67 @@
-            "lifecycle_stage='active', name='myname', tags={}>"
+            str(exp)
+            == "<Experiment: artifact_location='hi', creation_time=1661624204, experiment_id=0, "
+            "last_update_time=1661624204, lifecycle_stage='active', name='myname', tags={}>"
```
Do we have tests for `restore_experiment` and `delete_experiment`?

subramaniam02 replied:
There are no explicit checks for last_update_time in those scenarios. I am working on it.

subramaniam02 replied:
@harupy I am envisioning something like below. do think just checking `last_update_time != None` is good enough.

```
        exp1 = fs.get_experiment(exp_id)

        // ms might be too low, now() - 5 second ???
        exp1._set_last_update_time(int(time.time() * 1000))

        fs.restore_experiment(exp_id)
        self.assertEqual(fs.get_experiment(exp_id).name, new_name)
        assert restored_1.last_update_time > exp1.last_update_time
```

harupy replied:
Can we do this?

```python
        exp_id = fs.create_experiment("name")
        exp = fs.get_experiment(exp_id)
        time.sleep(0.01)  # 10 ms
        fs.restore_experiment(exp_id)
        assert fs.get_experiment(exp_id).last_update_time > exp.last_update_time
```

subramaniam02 replied:
I was thinking sleep might slow down tests. But given the number of test cases for experiments, I think it should be fine.

harupy replied:
A short sleep (< 10 ms) is fine, but a long sleep is not.

harupy commented on mlflow/store/tracking/file_store.py:
```diff
@@ -337,7 +337,14 @@ def _create_experiment_with_id(self, name, experiment_id, artifact_uri, tags):
         )
```
Can we also update `sqlalchemy_store.py`?

subramaniam02 replied:
I was thinking this would be handled since we have set sqlalchemy defaults.

harupy replied:
Does `SqlAlchemyStore.restore_experiment` automatically update the last_update_time column? I don't it does.

subramaniam02 replied:
I will check and add test cases for the same. I will make this as draft PR.

subramaniam02 replied:
you are right. I have added them.

harupy commented on mlflow/store/db_migrations/versions/97727af70f4d_creation_time_last_update_time_experiments.py:
```diff
@@ -0,0 +1,25 @@
+"""creation_time & last_update_time in experiments
```
```suggestion
"""Add creation_time and last_update_time to experiments table
```

harupy commented on mlflow/tracking/fluent.py:
```diff
@@ -1238,13 +1243,14 @@ def delete_experiment(experiment_id: str) -> None:
         Name: New Experiment
         Artifact Location: file:///.../mlruns/2
         Lifecycle_stage: deleted
+        Last Updated timestamp: 1661624204
```
```suggestion
        Last Updated timestamp: 1661624204
```

Is `1661624204` in milli seconds?

subramaniam02 replied:
no,  changed to millisecond

subramaniam02 committed:
case changes and tests using time.time() * 1000

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
case changes and tests using time.time() * 1000

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
Update mlflow/store/db_migrations/versions/97727af70f4d_creation_time_last_update_time_experiments.py

Co-authored-by: Harutaka Kawamura <hkawamura0130@gmail.com>
Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
Update mlflow/store/db_migrations/versions/97727af70f4d_creation_time_last_update_time_experiments.py

Co-authored-by: Harutaka Kawamura <hkawamura0130@gmail.com>
Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
test cases for last_update_time and creation_time added

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
test cases for last_update_time and creation_time added

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
keeping it same

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
keeping it same

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

harupy committed:
Return dummy object (#6701)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Fix flaky test_download_artifacts (#6702)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Temporarily use `tensorflow>=2.10.0rc3` in `keras>=2.10.0` (#6700)

* Use RC tensorflow

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix specifier

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Change base image to `python:3.7` in `Dockerfile` (#6699)

* Avoid changing the base environment

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix

Signed-off-by: harupy <hkawamura0130@gmail.com>

* use conda run

Signed-off-by: harupy <hkawamura0130@gmail.com>

* use python:3.7

Signed-off-by: harupy <hkawamura0130@gmail.com>

* use python:3.7

Signed-off-by: harupy <hkawamura0130@gmail.com>

* update dev/Dockerfile.test

Signed-off-by: harupy <hkawamura0130@gmail.com>

* use python:3.7

Signed-off-by: harupy <hkawamura0130@gmail.com>

* set CMD

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Remove `tabulate` (#6706)

* remove tabulate

Signed-off-by: harupy <hkawamura0130@gmail.com>

* type hint

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix default value of column_sep

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

subramaniam02 committed:
rerun tests

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
rerun tests

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

harupy committed:
Clean up imports in `mlflow/__init__.py` for more efficient code navigation (#6710)

* Clean up imports in mlflow/__init__.py

Signed-off-by: harupy <hkawamura0130@gmail.com>

* remove mlflow prefix

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Exclude examples/pipelines/sklearn_regression (#6711)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Add `.git-blame-ignore-revs` to ignores commits for `black` in git blame (#6704)

* Add .git-blame-ignore-revs to ignore black commit

Signed-off-by: harupy <hkawamura0130@gmail.com>

* add commit

Signed-off-by: harupy <hkawamura0130@gmail.com>

* capitalize

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Revert "Temporarily use `tensorflow>=2.10.0rc3` in `keras>=2.10.0` (#6700)" (#6717)

This reverts commit d0932a64e3602b754622533043b25f26b80e98db.

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Install miniforge in Dockerfile.test (#6708)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Update MLflow UI (#6666)

* Project import generated by Copybara.

GitOrigin-RevId: 9c923f71ea7a39cb1afde2eedc85cdf2d9557734
Signed-off-by: harupy <hkawamura0130@gmail.com>

* rename

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>
Signed-off-by: harupy <hkawamura0130@gmail.com>

* update content

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>
Signed-off-by: harupy <hkawamura0130@gmail.com>

* Reconcile changes from universe with OSS

Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>
Signed-off-by: harupy <hkawamura0130@gmail.com>

* delete yarn-3.2.1.cjs

Signed-off-by: harupy <hkawamura0130@gmail.com>

* run lint check on jsx,ts,tsx files

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>
Signed-off-by: harupy <17039389+harupy@users.noreply.github.com>
Co-authored-by: Richard Zang <Zangr@users.noreply.github.com>

dbczumar committed:
Fix CRAN incoming check (#6718)

* Fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Patch

Signed-off-by: dbczumar <corey.zumar@databricks.com>

Signed-off-by: dbczumar <corey.zumar@databricks.com>

harupy committed:
insert space (#6720)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

harupy committed:
Use monkeypatch.syspath_prepend when mutating sys.path (#6725)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

WeichenXu123 committed:
Remove DefaultEvaluator shap plot de-duplicating feature names logic and fix color bar (#6721)

* init

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* update

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* fix color bar

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* clean

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* clean

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* update

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

* fix

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

jerrylian-db committed:
Fix accuracy score feature name in model validation (#6729)

* wip

Signed-off-by: Jerry Liang <jerry.liang@databricks.com>

* wip

Signed-off-by: Jerry Liang <jerry.liang@databricks.com>

Signed-off-by: Jerry Liang <jerry.liang@databricks.com>

hubertzub-db committed:
fix: regression caused by #6376 (#6689)

* fix: regression caused by #6376

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: CR fix

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: CR fixes

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: fixed unit tests

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: fixed unit tests

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: removed artifact

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>

* chore: Update env variable value

Co-authored-by: Liang Zhang <liang.zhang@databricks.com>

Signed-off-by: Hubert Zub <hubert.zub@databricks.com>
Co-authored-by: Liang Zhang <liang.zhang@databricks.com>

gwy1995 committed:
Remove duplicate `_SPARK_MODEL_PATH_SUB` (#6683)

* #6682 Remove duplicate `_SPARK_MODEL_PATH_SUB`

Signed-off-by: zero <164434719@qq.com>

* update tests expectations

 The tests' expectations are incorrect due to duplicate _SPARK_MODEL_PATH_SUB #6683

Signed-off-by: zero <164434719@qq.com>

Signed-off-by: zero <164434719@qq.com>

dbczumar committed:
Scope Databricks inferred pip requirements to modules (#6722)

* Fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Tests

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Format

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Fix attempt

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Print

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Comment

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Debug

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Fix

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Explicit mapping of automl_runtime

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Test fixes and format

Signed-off-by: dbczumar <corey.zumar@databricks.com>

* Fix tests

Signed-off-by: dbczumar <corey.zumar@databricks.com>

Signed-off-by: dbczumar <corey.zumar@databricks.com>

kriscon-db committed:
Fix predict step target (#6690)

* Fix the makefile

Fix predict step target

Attempt to fix predict parquet read in spark

Remove fix

Another try

Fixed comment

Try to write the spark dataframe to the MLP cache

* Revert parquet write

* Black formatting

marijncv committed:
add sklearn usage example (#6694)

* add sklearn usage example

Signed-off-by: Marijn Valk <marijncv@hotmail.com>

* simplify scikit example

Signed-off-by: Marijn Valk <marijncv@hotmail.com>

Signed-off-by: Marijn Valk <marijncv@hotmail.com>

sunishsheth2009 committed:
Updating contributing to match the required version in generate-proto.sh (#6737)

Signed-off-by: Sunish Sheth <sunishsheth2009@gmail.com>

Signed-off-by: Sunish Sheth <sunishsheth2009@gmail.com>

harupy committed:
Add `pos_label` to `mlflow.evaluate` (#6696)

* Add pos_label to mlflow.evaluate

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix dataset_name

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix _get_classifier_per_class_metrics_collection_df

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix failed tests

Signed-off-by: harupy <hkawamura0130@gmail.com>

* use evaluator_config

Signed-off-by: harupy <hkawamura0130@gmail.com>

* remove average

Signed-off-by: harupy <hkawamura0130@gmail.com>

* revert

Signed-off-by: harupy <hkawamura0130@gmail.com>

* remove pos_label

Signed-off-by: harupy <hkawamura0130@gmail.com>

* revert changes in examples/pipelines/sklearn_regression

Signed-off-by: harupy <hkawamura0130@gmail.com>

* default value

Signed-off-by: harupy <hkawamura0130@gmail.com>

* add **

Signed-off-by: harupy <hkawamura0130@gmail.com>

* Update mlflow/models/evaluation/base.py

Co-authored-by: Corey Zumar <39497902+dbczumar@users.noreply.github.com>
Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix metrics to log in default evaluator

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix doc

Signed-off-by: harupy <hkawamura0130@gmail.com>

* mention default value

Signed-off-by: harupy <hkawamura0130@gmail.com>

* add _score

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix

Signed-off-by: harupy <hkawamura0130@gmail.com>

* add average and sett pos_label to 1 by default

Signed-off-by: harupy <hkawamura0130@gmail.com>

* remove _get_classifier_metrics

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix doc

Signed-off-by: harupy <hkawamura0130@gmail.com>

* refactor

Signed-off-by: harupy <hkawamura0130@gmail.com>

* add test

Signed-off-by: harupy <hkawamura0130@gmail.com>

* rename dataset

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix

Signed-off-by: harupy <hkawamura0130@gmail.com>

* test pos_label=None

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix ternary

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix

Signed-off-by: harupy <hkawamura0130@gmail.com>

* fix tests

Signed-off-by: harupy <hkawamura0130@gmail.com>

* remove redundant function parameters

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>
Co-authored-by: Corey Zumar <39497902+dbczumar@users.noreply.github.com>

harupy committed:
Remove useless unused-variable (#6741)

Signed-off-by: harupy <hkawamura0130@gmail.com>

Signed-off-by: harupy <hkawamura0130@gmail.com>

subramaniam02 commented:
@harupy Can you review this PR once again ?

harupy commented on tests/store/tracking/test_file_store.py:
```diff
@@ -402,10 +408,14 @@ def test_create_experiment(self):
             exp1.artifact_location,
             path_to_local_file_uri(posixpath.join(self.test_root, created_id)),
         )
+        assert exp1.creation_time
```
```suggestion
        assert exp1.creation_time is None
```

We should avoid assertion like this because it's unclear what is checked.

subramaniam02 committed:
rebasing master

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 committed:
revert rebasing master

Signed-off-by: Subramaniam Shanmugam <subramaniam02@gmail.com>

subramaniam02 commented:
Screwed up some git ops. Will open a new PR.
